init-common/prepare-dvm-sysvinit.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Source Qubes library.
+. /usr/lib/qubes/init-common/functions
+
+possibly_run_save_script
+umount /rw
+dmesg -c >/dev/null
+free | grep Mem: | (read a b c d ; qubesdb-write /qubes-used-mem $c)
+# give dom0 time to read some entries, when done it will shutdown qubesdb,
+# so wait for it
+qubesdb-watch /stop-qubesdb
+# just to make sure
+systemctl stop qubes-db.service
+
+# we're still running in DispVM template
+echo "Waiting for save/restore..."
+# the service will start only after successful restore
+systemctl start qubes-db.service
+echo Back to life.
+
+# Reload random seed
+qubesdb-read /qubes-random-seed | base64 -d > /dev/urandom
+qubesdb-rm /qubes-random-seed
```

Comment:
This file doesn't look like sysvinit version - it calls systemctl. I'd drop it and the same `prepare-dvm.sh`. The differences in behavior are unintentional (sysvinit version wasn't updated). The "systemd" version seems to be better. If someone would like to restore sysvinit support, the only thing needed should be replacing `systemctl` with `service`.